src/components/editor/faultTree/Editor.tsx

```diff
@@ -119,6 +119,33 @@ const Editor = () => {
     }
   };
 
+  const handleAutomaticLayout = async (graph, jointPaper) => {
+    for (const el of graph.getElements()) {
+      const faultEventIri = el.get(JOINTJS_NODE_MODEL.faultEventIri);
+      const movedEvent = findEventByIri(faultEventIri, getRootEvent());
+
+      if (movedEvent) {
+        const elementView = el.findView(jointPaper);
+        const size = elementView.model.attributes.size;
+        const position = elementView.model.attributes.position;
+        const rect: Rectangle = movedEvent.rectangle;
+
+        if (rect.x != position.x || rect.y != position.y || rect.width != size.width || rect.height != size.height) {
+          rect.x = position.x;
+          rect.y = position.y;
+          rect.width = size.width;
+          rect.height = size.height;
+
+          try {
+            await faultEventService.updateEventRectangle(faultEventIri, rect.iri, rect);
```

Comment:
@kostobog as I assume there is no way to update several nodes simultaneously, I just made an async/await to treat them one by one, when we persist several nodes due to automatic layout.

Reply:
Will be resolved by https://github.com/kbss-cvut/fta-fmea/issues/160